diff --git a/spring/rts/Rendering/GL/myGL.cpp b/spring/rts/Rendering/GL/myGL.cpp
index de512256c..b527e1b27 100644
--- a/spring/rts/Rendering/GL/myGL.cpp
+++ b/spring/rts/Rendering/GL/myGL.cpp
@@ -146,8 +146,8 @@ static bool GetVideoMemInfoMESA(GLint* memInfo)
 
 	typedef PFNGLXQUERYCURRENTRENDERERINTEGERMESAPROC QCRIProc;
 
-	static constexpr const GLubyte* qcriProcName = (const GLubyte*) "glXQueryCurrentRendererIntegerMESA";
-	static           const QCRIProc qcriProcAddr = (QCRIProc) glXGetProcAddress(qcriProcName);
+	static const GLubyte* qcriProcName = (const GLubyte*) "glXQueryCurrentRendererIntegerMESA";
+	static const QCRIProc qcriProcAddr = (QCRIProc) glXGetProcAddress(qcriProcName);
 
 	if (qcriProcAddr == nullptr)
 		return false;
