diff --git a/rts/Rendering/GLContext.h b/rts/Rendering/GLContext.h
index f44d1d1..fa38832 100644
--- a/rts/Rendering/GLContext.h
+++ b/rts/Rendering/GLContext.h
@@ -26,6 +26,8 @@
 
 // TODO: add event driven system (like alttab, resolution changed, dualscreen, ...)
 
+#include <stddef.h>
+
 namespace GLContext
 {
 	typedef void (*Func)(void* data);
diff --git a/rts/System/MemPool.h b/rts/System/MemPool.h
index cef4947..558e04b 100644
--- a/rts/System/MemPool.h
+++ b/rts/System/MemPool.h
@@ -1,6 +1,7 @@
 #ifndef CMEMPOOL_H
 #define CMEMPOOL_H
 
+#include <stddef.h>
 #include <new>
 
 const size_t MAX_MEM_SIZE=200;
