diff --git a/rts/Rendering/Models/AssIO.h b/rts/Rendering/Models/AssIO.h
index 16b726078d..b899e57faf 100644
--- a/rts/Rendering/Models/AssIO.h
+++ b/rts/Rendering/Models/AssIO.h
@@ -3,8 +3,8 @@
 #ifndef ASS_IO_H
 #define ASS_IO_H
 
-#include "lib/assimp/include/assimp/IOStream.hpp"
-#include "lib/assimp/include/assimp/IOSystem.hpp"
+#include <assimp/IOStream.hpp>
+#include <assimp/IOSystem.hpp>
 class CFileHandler;
 
 // Custom implementation of Assimp IOStream to support Spring's VFS
diff --git a/rts/Rendering/Models/AssParser.cpp b/rts/Rendering/Models/AssParser.cpp
index 121d9d5009..26070f24f5 100644
--- a/rts/Rendering/Models/AssParser.cpp
+++ b/rts/Rendering/Models/AssParser.cpp
@@ -16,13 +16,13 @@
 #include "System/FileSystem/FileHandler.h"
 #include "System/FileSystem/FileSystem.h"
 
-#include "lib/assimp/include/assimp/config.h"
-#include "lib/assimp/include/assimp/defs.h"
-#include "lib/assimp/include/assimp/types.h"
-#include "lib/assimp/include/assimp/scene.h"
-#include "lib/assimp/include/assimp/postprocess.h"
-#include "lib/assimp/include/assimp/Importer.hpp"
-#include "lib/assimp/include/assimp/DefaultLogger.hpp"
+#include <assimp/config.h>
+#include <assimp/defs.h>
+#include <assimp/types.h>
+#include <assimp/scene.h>
+#include <assimp/postprocess.h>
+#include <assimp/Importer.hpp>
+#include <assimp/DefaultLogger.hpp>
 
 #include <regex>
 
diff --git a/rts/Rendering/Models/IModelParser.cpp b/rts/Rendering/Models/IModelParser.cpp
index fddb424f8a..27f28b24ff 100644
--- a/rts/Rendering/Models/IModelParser.cpp
+++ b/rts/Rendering/Models/IModelParser.cpp
@@ -16,7 +16,7 @@
 #include "System/MainDefines.h" // SNPRINTF
 #include "System/SafeUtil.h"
 #include "System/Threading/ThreadPool.h"
-#include "lib/assimp/include/assimp/Importer.hpp"
+#include <assimp/Importer.hpp>
 
 
 CModelLoader modelLoader;
diff --git a/rts/builds/dedicated/CMakeLists.txt b/rts/builds/dedicated/CMakeLists.txt
index 05f1e93c5f..7a120202f9 100644
--- a/rts/builds/dedicated/CMakeLists.txt
+++ b/rts/builds/dedicated/CMakeLists.txt
@@ -20,8 +20,10 @@ set(ENGINE_SRC_ROOT_DIR "${CMAKE_SOURCE_DIR}/rts")
 
 
 ### Assemble libraries
+find_package(ZLIB REQUIRED)
+find_package(MiniZip REQUIRED)
 list(APPEND engineDedicatedLibraries ${Boost_REGEX_LIBRARY})
-list(APPEND engineDedicatedLibraries lua archives 7zip ${SPRING_MINIZIP_LIBRARY} ${ZLIB_LIBRARY} gflags)
+list(APPEND engineDedicatedLibraries lua archives 7zip ${MINIZIP_LIBRARY} ${ZLIB_LIBRARY} gflags)
 list(APPEND engineDedicatedLibraries headlessStubs engineSystemNet)
 list(APPEND engineDedicatedLibraries ${LIBUNWIND_LIBRARIES})
 
diff --git a/rts/builds/headless/CMakeLists.txt b/rts/builds/headless/CMakeLists.txt
index e73b62ed6c..cbd27fd350 100644
--- a/rts/builds/headless/CMakeLists.txt
+++ b/rts/builds/headless/CMakeLists.txt
@@ -43,6 +43,9 @@ list(APPEND engineHeadlessLibraries no-sound)
 list(APPEND engineHeadlessLibraries engineSim)
 list(APPEND engineHeadlessLibraries pr-downloader_static)
 
+find_package(ZLIB REQUIRED)
+list(APPEND engineHeadlessLibraries ${ZLIB_LIBRARY})
+
 include_directories(${ENGINE_SRC_ROOT_DIR}/lib/assimp/include)
 include_directories(${ENGINE_SRC_ROOT_DIR}/lib/asio/include)
 include_directories(${ENGINE_SRC_ROOT_DIR}/lib/slimsig/include)
diff --git a/rts/lib/CMakeLists.txt b/rts/lib/CMakeLists.txt
index 04b3887b9d..426b2d8f73 100644
--- a/rts/lib/CMakeLists.txt
+++ b/rts/lib/CMakeLists.txt
@@ -4,7 +4,6 @@ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/rts)
 ADD_SUBDIRECTORY(lua)
 ADD_SUBDIRECTORY(luasocket)
 ADD_SUBDIRECTORY(7z)
-ADD_SUBDIRECTORY(minizip)
 ADD_SUBDIRECTORY(headlessStubs)
 if (ENABLE_STREFLOP)
 ADD_SUBDIRECTORY(streflop)
@@ -60,9 +59,6 @@ foreach(var ASSIMP_BUILD_STATIC_LIB ASSIMP_NO_EXPORT)
 	set("${var}" ON CACHE STRING "forced on by spring build env" FORCE)
 endforeach()
 
-ADD_SUBDIRECTORY(assimp)
-target_compile_definitions(assimp PRIVATE -DASSIMP_BUILD_NO_OWN_ZLIB)
-
 if    (NOT HEADLESS_SYSTEM)
 
 	if    (USE_LIBSQUISH)
diff --git a/tools/unitsync/CMakeLists.txt b/tools/unitsync/CMakeLists.txt
index f46a3ca17a..d3a00445ed 100644
--- a/tools/unitsync/CMakeLists.txt
+++ b/tools/unitsync/CMakeLists.txt
@@ -8,11 +8,12 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "../..")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
 
 find_package_static(ZLIB REQUIRED)
+find_package_static(MiniZip REQUIRED)
 
 list(APPEND unitsync_libs ${CMAKE_DL_LIBS})
 list(APPEND unitsync_libs 7zip lua headlessStubs archives)
 list(APPEND unitsync_libs ${ZLIB_LIBRARY})
-list(APPEND unitsync_libs ${SPRING_MINIZIP_LIBRARY})
+list(APPEND unitsync_libs ${MINIZIP_LIBRARY})
 
 if (WIN32)
 	list(APPEND unitsync_libs ${IPHLPAPI_LIBRARY})
@@ -46,7 +47,7 @@ remove_definitions(-DTHREADPOOL)
 
 set(ENGINE_SRC_ROOT "../../rts")
 
-include_directories(${SPRING_MINIZIP_INCLUDE_DIR})
+include_directories(${MINIZIP_INCLUDE_DIR})
 include_directories(${ENGINE_SRC_ROOT}/lib/lua/include)
 include_directories(${ENGINE_SRC_ROOT}/lib/7zip)
 include_directories(${ENGINE_SRC_ROOT})
